ls1intum/Artemis
# [Refactoring] Export student submissions

### Checklist

### Motivation and Context


At the moment, there is code duplication within the export student submissions functions for Text-, File-, and Modeling exercises.

### Description

This PR refactors the code and removes the duplicated code.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a Text-, Modeling-, and File Upload Exercise
4. Submit to all three exercises
5. Navigate to the detail views of the exercises and press "export submissions"
6. Download the zip file, open it, and make sure that the submitted answer is inside

### Test Coverage



FileService.java 82%
SubmissionExportService.java 89%
FileUploadSubmissionResource.java 80%
ModelingExerciseResource.java 87%
TextExerciseResource.java 90%
ResponseUtil.java 94%
